packages/liveblocks-yjs/src/awareness.ts

```diff
@@ -0,0 +1,125 @@
+// TODO: apparently Yjs is full of anys or something, see if we can fix this
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import type {
+  BaseUserMeta,
+  Json,
+  JsonObject,
+  LsonObject,
+  Room,
+} from "@liveblocks/client";
+import { Observable } from "lib0/observable";
+import type * as Y from "yjs";
+
+const Y_PRESENCE_KEY = "__yjs";
+
+type MetaClientState = {
+  clock: number;
+  lastUpdated: number;
+};
+
+/**
+ * This class will store Yjs awareness in Liveblock's presence under the __yjs key
+ * IMPORTANT: The Yjs awareness protocol uses ydoc.clientId to reference users
+ * to their respective documents. To avoid mapping Yjs clientIds to liveblock's connectionId,
+ * we simply set the clientId of the doc to the connectionId. Then no further mapping is required
+ */
+export class Awareness extends Observable<unknown> {
```

Comment:
Didn't read this code closely, because I assume they're just code moves? Lmk if there are any changes in here I should pay attention to.